clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -405,8 +405,8 @@ private void processAssignmentReceived(ConsumerGroupHeartbeatResponseData.Assign
                 assignmentUnresolved, currentAssignment);
             // Make sure we transition the member back to STABLE if it was RECONCILING (ex.
             // member was RECONCILING unresolved assignments that were just removed by the
-            // broker).
-            if (state == MemberState.RECONCILING) {
+            // broker), or JOINING (member joining received empty assignment).
+            if (state == MemberState.RECONCILING || state == MemberState.JOINING) {
                 // This is the case where a member was RECONCILING an unresolved
                 // assignment that was removed by the broker in a following assignment.
```

Comment:
This comment is not completely correct anymore. It only describes one case of the two. Doesn't this comment  duplicate the comment above the `if`-statement.
I really strongly recommend to not use too many inline comments. They are really hard to properly maintain as we can see here.

Reply:
You're right, the comment is a dup of the one above and was not accurate anymore, removed (agree with the suggestion of not overdoing the inline comments, noted)